### ISSUE TITLE
🐞Flutter Engine stops when workmanager used with Google_Mobile_ads

### ISSUE THREAD

**Issue by agacemi** [fluttercommunity/flutter_workmanager#347]
- [x] I have read the [README](https://github.com/vrtdev/flutter_workmanager/blob/master/README.md)
- [x] I have done the setup for [Android](https://github.com/vrtdev/flutter_workmanager/blob/master/ANDROID_SETUP.md)
- [x] I have done the setup for [iOS](https://github.com/vrtdev/flutter_workmanager/blob/master/IOS_SETUP.md)
- [x] I have ran the sample app and it does not work there

**Version**

| Technology           | Version       |
| -------------------  | ------------- |
| Workmanager version  |    0.4.1           |

**Describe the error**

In our project, we use workmanager and google_mobile_ads and this seems causes a critical display issue. In fact, the flutter engine is stopped by the library. Please, you can find sample on this repo https://github.com/agacemi/sample-workmanager-ads-issue 

**How to reproduce the issue**

- Click on "Start the Flutter background service"
- Click on "Register OneOff Task
- Wait for receiving the success notification
- Scoll or click on Fake button
 
This video shows this behavior : 

https://user-images.githubusercontent.com/5896654/143801545-34597f47-5b48-4fe8-b3b3-f8d3867212bc.mp4

**Output of `flutter doctor -v`**

[✓] Flutter (Channel stable, 2.5.3, on Mac OS X 10.15.7 19H1417 darwin-x64, locale fr-FR)
    • Flutter version 2.5.3 at /Users/abdelaziz_gacemi/Devhome/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 18116933e7 (6 weeks ago), 2021-10-15 10:46:35 -0700
    • Engine revision d3ea636dc5
    • Dart version 2.14.4

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.4, Build version 12D4e
    • CocoaPods version 1.11.2

**Comment by ened** [comment]
@agacemi I'm unable to reproduce this issue when downloading your repo and running it on a Pixel 4. Please take detail logs (pidcat) and share them here. There are also no ads loading for me.

Also please retest the issue using the pre-release version of `workmanager`:
```
  workmanager: ^0.5.0-dev.7
```

**Comment by agacemi** [comment]
Thank you @ened  for investigating the issue.  I tested with ^0.5.0-dev.7 and I have the same issue. 

Indeed, if ad fails to load, then the issue can't reproduced. May be an internal implementation choice of google mobile ads, associated with workmanager that throw the issue.  

Please, try Nexus 6 emulator. I reproduced the issue with bit different behavior (the app crashs).

<img width="992" alt="Capture d’écran 2021-12-07 à 14 48 26" src="https://user-images.githubusercontent.com/5896654/145040869-df8ad3c9-2254-4472-964f-638d9d5048c2.png">
 
For information, when we comment the fellowing line, the app 'works' well ... https://github.com/fluttercommunity/flutter_workmanager/blob/a65356d8bb3aa19b63c7ec800d94379f6df80492/android/src/main/kotlin/be/tramckrijte/workmanager/BackgroundWorker.kt#L137

Please, find below logs when app crash on emulator. 
>                         
                    Ads  W  #004 The webview is destroyed. Ignoring action.
                 chatty  I  uid=10154(dev.fluttercommunity.workmanager.example) identical 1 line
                    Ads  W  #004 The webview is destroyed. Ignoring action.
                flutter  I  ca-app-pub-3940256099942544/6300978111 :Loading Ad: width: 320, height: 100,
                    Ads  I  This request is sent from a test device.
               chromium  E  [ERROR:cookie_manager.cc(137)] Strict Secure Cookie policy does not allow setting a secure cookie for http://googleads.g.doubleclick.net/ for ap
                            ps targeting >= R. Please either use the 'https:' scheme for this URL or omit the 'Secure' directive in the cookie value.
                    Ads  W  #004 The webview is destroyed. Ignoring action.
                 chatty  I  uid=10154(dev.fluttercommunity.workmanager.example) identical 1 line
                    Ads  W  #004 The webview is destroyed. Ignoring action.
                flutter  I  ca-app-pub-3940256099942544/6300978111 :Loading Ad: width: 320, height: 100,
                    Ads  I  This request is sent from a test device.
               chromium  E  [ERROR:cookie_manager.cc(137)] Strict Secure Cookie policy does not allow setting a secure cookie for http://googleads.g.doubleclick.net/ for ap
                            ps targeting >= R. Please either use the 'https:' scheme for this URL or omit the 'Secure' directive in the cookie value.
                    Ads  W  #004 The webview is destroyed. Ignoring action.
                 chatty  I  uid=10154(dev.fluttercommunity.workmanager.example) identical 1 line
                    Ads  W  #004 The webview is destroyed. Ignoring action.
                flutter  I  ca-app-pub-3940256099942544/6300978111 :Loading Ad: width: 320, height: 100,
                    Ads  I  This request is sent from a test device.
               chromium  E  [ERROR:cookie_manager.cc(137)] Strict Secure Cookie policy does not allow setting a secure cookie for http://googleads.g.doubleclick.net/ for ap
                            ps targeting >= R. Please either use the 'https:' scheme for this URL or omit the 'Secure' directive in the cookie value.
                    Ads  W  #004 The webview is destroyed. Ignoring action.
                 chatty  I  uid=10154(dev.fluttercommunity.workmanager.example) identical 1 line
                    Ads  W  #004 The webview is destroyed. Ignoring action.
                flutter  I  ca-app-pub-3940256099942544/6300978111 :Loading Ad: width: 320, height: 100,
         manager.exampl  I  Waiting for a blocking GC ProfileSaver
                         I  WaitForGcToComplete blocked ProfileSaver on HeapTrim for 65.647ms
          EGL_emulation  D  eglCreateContext: 0xf2faff00: maj 3 min 0 rcv 3
                         D  eglCreateContext: 0xf2f97c40: maj 3 min 0 rcv 3
         HostConnection  D  HostConnection::get() New Host Connection established 0xf2fb06e0, tid 9001
                         D  HostComposition ext ANDROID_EMU_CHECKSUM_HELPER_v1 ANDROID_EMU_native_sync_v2 ANDROID_EMU_native_sync_v3 ANDROID_EMU_native_sync_v4 ANDROID_EMU_
                            dma_v1 ANDROID_EMU_direct_mem ANDROID_EMU_host_composition_v1 ANDROID_EMU_host_composition_v2 ANDROID_EMU_YUV_Cache ANDROID_EMU_async_unmap_buff
                            er ANDROID_EMU_has_shared_slots_host_memory_allocator ANDROID_EMU_sync_buffer_data GL_OES_EGL_image_external_essl3 GL_OES_vertex_array_object GL
                            _KHR_texture_compression_astc_ldr ANDROID_EMU_host_side_tracing ANDROID_EMU_async_frame_commands ANDROID_EMU_gles_max_version_3_0
          EGL_emulation  D  eglMakeCurrent: 0xf2f97c40: ver 3 0 (tinfo 0xb34e46d0) (first time)
     PathProviderPlugin  D  Don't use TaskQueues.
             FlutterJNI  W  FlutterJNI.setRefreshRateFPS called more than once
                         W  FlutterJNI.loadLibrary called more than once
                         W  FlutterJNI.prefetchDefaultFontManager called more than once
      ResourceExtractor  I  Found extracted resources res_timestamp-1-1638878765748
             FlutterJNI  W  FlutterJNI.init called more than once
         manager.exampl  I  NativeAlloc concurrent copying GC freed 12558(736KB) AllocSpace objects, 0(0B) LOS objects, 49% free, 4414KB/8829KB, paused 46us total 245.795ms
                         I  NativeAlloc concurrent copying GC freed 9463(631KB) AllocSpace objects, 0(0B) LOS objects, 49% free, 4330KB/8661KB, paused 26us total 113.817ms
         HostConnection  D  HostConnection::get() New Host Connection established 0xf2f98c70, tid 9000
                         D  HostComposition ext ANDROID_EMU_CHECKSUM_HELPER_v1 ANDROID_EMU_native_sync_v2 ANDROID_EMU_native_sync_v3 ANDROID_EMU_native_sync_v4 ANDROID_EMU_
                            dma_v1 ANDROID_EMU_direct_mem ANDROID_EMU_host_composition_v1 ANDROID_EMU_host_composition_v2 ANDROID_EMU_YUV_Cache ANDROID_EMU_async_unmap_buff
                            er ANDROID_EMU_has_shared_slots_host_memory_allocator ANDROID_EMU_sync_buffer_data GL_OES_EGL_image_external_essl3 GL_OES_vertex_array_object GL
                            _KHR_texture_compression_astc_ldr ANDROID_EMU_host_side_tracing ANDROID_EMU_async_frame_commands ANDROID_EMU_gles_max_version_3_0
          EGL_emulation  D  eglMakeCurrent: 0xf2faff00: ver 3 0 (tinfo 0xbb4c9370) (first time)
                         D  eglCreateContext: 0xf2fb7830: maj 3 min 0 rcv 3
                flutter  I  ca-app-pub-3940256099942544/6300978111 :Loading Ad: width: 320, height: 100,
                         I  simpleTask was executed. inputData = {int: 1, bool: true, double: 1.0, string: string, array: [1, 2, 3]}
                         I  Bool from prefs: true
       WM-WorkerWrapper  I  Worker result SUCCESS for Work [ id=3f669975-52d8-4dbc-8fab-66fdc443008f, tags={ be.tramckrijte.workmanager.BackgroundWorker } ]
        emuglGLESv2_enc  E  device/generic/goldfish-opengl/system/GLESv2_enc/GL2Encoder.cpp:s_glBufferData:578 GL error 0x502
                         E  device/generic/goldfish-opengl/system/GLESv2_enc/GL2Encoder.cpp:s_glUseProgram:2027 GL error 0x501
                         D  sendVertexAttributes: bad offset / len!!!!!
                   libc  F  Fatal signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0x8 in tid 8511 (manager.example), pid 8511 (manager.example)
                  DEBUG  F  #00 pc 0005657b  /apex/com.android.runtime/lib/bionic/libc.so (memmove_generic+651) (BuildId: 6e3a0180fa6637b68c0d181c343e6806)
                         F  #01 pc 00000008  <unknown>

**Comment by ened** [comment]
Interesting, @agacemi. I think the issue may be due to `workmanager` registering a Flutter engine and the mobile google ads SDK running + registering a view factory. When the `workmanager` engine is stopped, it may take a few things with it, including a unrelated view factory. (I do not have a better explanation - yet). Suspicious lines here https://github.com/googleads/googleads-mobile-flutter/blob/master/packages/google_mobile_ads/android/src/main/java/io/flutter/plugins/googlemobileads/GoogleMobileAdsPlugin.java#L240 .

My next attempt would be to prevent the `workmanager` engine from loading the mobile_ads_sdk plugin.

**Comment by agacemi** [comment]
@ened  Do you think that a fix is quite complex with regression risk?

**Comment by agacemi** [comment]
@ened  I have some news I would to share with you. 

- Actually, it is not a workmanager bug but flutter bug. It has been fixed in Flutter 2.5.0-7.0.pre.154 and the fix is available in the last release 2.8.0 (see https://docs.flutter.dev/development/tools/sdk/release-notes/release-notes-2.8.0, line [88455](https://github.com/flutter/flutter/pull/88455) Reland remove DefaultShaderWarmup).

 
- By git bisecting, I found the commit (   https://github.com/flutter/flutter/commit/c02961256a7eefc3c1404c1c28b927ef4cb78ef1) that fix the issue (freeze/crash). It isn't clear for me to understand the relationship between the fix and the behavior of flutterEngine.destory that causes the issue when Workmanager is associated with google_mobile_ads.  

- The associated pull request https://github.com/flutter/flutter/pull/88455 that explains the content.

Thank you for investigating the issue. I propose de close the issue
